modules/plugins/wfly-10/src/main/java/org/rhq/modules/plugins/wildfly10/NamingComponent.java

```diff
@@ -17,6 +27,48 @@
  * @author Heiko W. Rupp
  */
 public class NamingComponent extends BaseComponent implements OperationFacet {
+    private static final String BINDING_TYPE = "Binding";
+    private static final String ENVIRONMENT = "environment";
+    @Override
+    public CreateResourceReport createResource(CreateResourceReport report) {
+        if (report.getResourceType().getName().equals(BINDING_TYPE)) {
+            Configuration resourceConfiguration = report.getResourceConfiguration();
+            PropertyList prop = (PropertyList)resourceConfiguration.get(ENVIRONMENT);
```

Comment:
Actually, same to all of these ``.get()`` are all of these always present with the ``BINDING_TYPE`` or can something be null also? There's no null checks in this part of the code.